File: juliaos-core/python/src/juliaos/leverage/tests/test_leverage_system.py
"""
Tests for the Enhanced Leverage System
====================================

Comprehensive test suite covering core functionality,
induction engine, and integration features.
"""

import unittest
from pathlib import Path
from juliaos.leverage import (
    LeverageInductionEngine,
    run_intelligent_induction,
    quick_induction_analysis,
    get_induction_recommendations,
    health_check,
    scan_my_app,
    leverage_my_app
)

class TestLeverageSystem(unittest.TestCase):
    """Test suite for the Enhanced Leverage System"""
    
    def setUp(self):
        """Set up test environment"""
        self.test_project_path = str(Path(__file__).parent / "test_project")
        self.engine = LeverageInductionEngine()
    
    def test_induction_engine_initialization(self):
        """Test that the induction engine initializes correctly"""
        self.assertIsNotNone(self.engine)
        self.assertEqual(self.engine.project_path, ".")
    
    def test_quick_analysis(self):
        """Test quick analysis functionality"""
        analysis = quick_induction_analysis(self.test_project_path)
        self.assertIsInstance(analysis, dict)
        self.assertIn('project_analysis', analysis)
        self.assertIn('basic_strategy', analysis)
    
    def test_recommendations(self):
        """Test recommendation generation"""
        recommendations = get_induction_recommendations(self.test_project_path)
        self.assertIsInstance(recommendations, list)
        self.assertTrue(len(recommendations) > 0)
    
    def test_health_check(self):
        """Test system health check"""
        health = health_check(self.test_project_path)
        self.assertIsInstance(health, dict)
        self.assertIn('status', health)
        self.assertIn('recommendations', health)
    
    def test_app_scan(self):
        """Test application scanning"""
        scan = scan_my_app(self.test_project_path)
        self.assertIsInstance(scan, dict)
        self.assertIn('services', scan)
        self.assertIn('ready', scan)
    
    def test_feature_leverage(self):
        """Test feature-specific leverage"""
        result = leverage_my_app("test_feature", self.test_project_path)
        self.assertIsInstance(result, dict)
        self.assertIn('exponential_value', result)
        self.assertIn('business_value', result)
        self.assertIn('implementation_effort', result)
    
    def test_intelligent_induction(self):
        """Test complete induction process"""
        results = run_intelligent_induction(self.test_project_path)
        self.assertIsInstance(results, dict)
        self.assertIn('project_analysis', results)
        self.assertIn('leverage_strategy', results)
        self.assertIn('implementation_plan', results)

if __name__ == '__main__':
    unittest.main()

File: juliaos-core/julia/test/leverage/test_leverage_system.jl
"""
Tests for the Enhanced Leverage System Julia Bridge
================================================

Test suite for the Julia-Python bridge functionality.
"""

using Test
using JuliaOS.LeverageSystem

@testset "LeverageSystem" begin
    test_project_path = joinpath(@__DIR__, "test_project")
    
    @testset "Induction Process" begin
        # Test quick analysis
        analysis = quick_induction_analysis(test_project_path)
        @test analysis isa Dict
        @test haskey(analysis, "project_analysis")
        @test haskey(analysis, "basic_strategy")
        
        # Test recommendations
        recommendations = get_induction_recommendations(test_project_path)
        @test recommendations isa Vector{String}
        @test length(recommendations) > 0
        
        # Test complete induction
        results = run_intelligent_induction(test_project_path)
        @test results isa Dict
        @test haskey(results, "project_analysis")
        @test haskey(results, "leverage_strategy")
        @test haskey(results, "implementation_plan")
    end
    
    @testset "System Operations" begin
        # Test health check
        health = health_check(test_project_path)
        @test health isa Dict
        @test haskey(health, "status")
        @test haskey(health, "recommendations")
        
        # Test app scanning
        scan = scan_my_app(test_project_path)
        @test scan isa Dict
        @test haskey(scan, "services")
        @test haskey(scan, "ready")
        
        # Test feature leverage
        result = leverage_my_app("test_feature", test_project_path)
        @test result isa Dict
        @test haskey(result, "exponential_value")
        @test haskey(result, "business_value")
        @test haskey(result, "implementation_effort")
    end
    
    @testset "Report Generation" begin
        # Test report generation
        report = generate_leverage_report(test_project_path)
        @test report isa String
        @test length(report) > 0
    end
end